src/buildSummary.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The MathWorks, Inc.
+import * as core from "@actions/core";
+import { join } from 'path';
+import { readFile } from 'fs';
+import { promisify } from 'util';
+
+// Promisify the readFile function to use it with async/await
+const readFileAsync = promisify(readFile);
+
+export interface Task {
+  name: string;
+  description: string;
+  failed: boolean;
+  skipped: boolean;
+  duration: string;
+}
+
+export interface TaskList {
+  taskDetails: Task[];
+}
```

Comment:
I don't see the need for this wrapper type.

Reply:
Done.